magenta-lib/src/test/scala/magenta/deployment_type/ParamTest.scala

```diff
@@ -1,20 +1,31 @@
 package magenta.deployment_type
 
 import java.util.UUID
-
 import magenta.artifact.S3Path
 import magenta._
 import magenta.fixtures._
 import org.scalatest.flatspec.AnyFlatSpec
 import org.scalatest.matchers.should.Matchers
 import org.mockito.MockitoSugar
-import play.api.libs.json.JsString
+import play.api.libs.json.{JsNumber, JsString, JsValue}
 
+import java.time.Duration.ofSeconds
 import scala.collection.mutable
 
-class TestRegister extends ParamRegister {
-  val paramsList = mutable.Map.empty[String, Param[_]]
-  def add(param: Param[_]) = paramsList += param.name -> param
+class ParamRegistrationTest extends AnyFlatSpec with Matchers {
+  class TestRegister extends ParamRegister {
+    val paramsList = mutable.Map.empty[String, Param[_]]
+    def add(param: Param[_]) = paramsList += param.name -> param
+  }
+
+  "Param registration" should "occur when a param is created" in {
+    implicit val register: TestRegister = new TestRegister
```

Comment:
Separating this `ParamRegistrationTest` class out lets us give a specific `TestRegister` to the one test that cares about it, and give a single shared stub `ParamRegister` to all the other tests that don't care about it all - this removes this repetitive line of noise from each test case:

https://github.com/guardian/riff-raff/blob/192e63ff5ba27382d3b2620185c7c817e6fae471/magenta-lib/src/test/scala/magenta/deployment_type/ParamTest.scala#L45